controllers/asset_controller.go

```diff
@@ -74,7 +76,12 @@ func (r *AssetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 	if err := r.Get(ctx, req.NamespacedName, asset); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
-	assetVersion := checksum(asset.Spec.EncryptedData)
+	assetVersion := checksum(mergeMap(
```

Comment:
Merging these maps into a new map may not be a good idea. Some values will be dropped during the merge if there are duplicated keys among these maps.

I think a better way to do this is summing up each checksum from each map.

Reply:
You are right, especially annotations and labels could easily conflict.

How about convert every map to a sorted array with item in '<key>:<value>' format then send them into hash function?

Also I'm wondering why choose crc32 instead of other hash algorithm?

Reply:
> How about convert every map to a sorted array with item in ':' format then send them into hash function?

It is ok as long as all keys and values are counted into the hash.

> Also I'm wondering the reason about original implementation choose crc32 instead of other hash algorithm?

No special reason. A quick hash is enough and the crc32 is built in the standard library so that no additional dependency is required.

Reply:
I've change checksum function to use `client.Object` as input, marshal to json, then calculate crc32.
And pass field we want to check into the function.